src/chebyshev_mod.F90

```diff
@@ -0,0 +1,290 @@
+!> Module to obtain the density matrix by applying a Chebyshev
+!! polynomial expansion.
+!!
+!! \ingroup PROGRESS
+!!
+!!  See Amparo Gil 2007 \cite Amparo2007 ,
+!!  See Silver et al \cite Silver1996 ,
+!!  See Weisse et al \cite Weisse2006
+!!
+!! \toodo Add the power method in BML to get a better estimate of the spectral boundaries.
+!!
+module Chebyshev_mod
+
+  use bml
+  use normalize_mod
+  use densitymatrix_mod
+  use openfiles_mod
+
+  implicit none
+
+  private  !Everything is private by default
+
+  integer, parameter :: dp = kind(1.0d0)
+
+  public :: build_density_cheb
+
+contains
+
+  !> Builds the density matrix from \f$ H_0 \f$ for a Fermi function
+  !! approximated with a Chebyshev polynomial expansion.
+  !!
+  !! \f$ \rho_{n+1} = b_{n+1}T_{n+1} + \rho_{n} \f$
+  !! Where,\f$ T_n \f$ is the nth Chebyshev polynomial and
+  !! \f$ b_{n} \f$  is the nth coefficient of the expansion for the Fermi function.
+  !! \param ham_bml Input Orthogonalized Hamiltonian matrix.
+  !! \param rho_bml Output density matrix,
+  !! \param threshold Threshold for sparse matrix algebra.
+  !! \param ncoeffs Number of Chebyshev coefficients.
+  !! \param kbt Electronic temperature in the energy units of the Hamiltonian.
+  !! \param ef Fermi level in the energy units of the Hamiltonian.
+  !! \param verbose Verbosity level.
+  !!
+  subroutine build_density_cheb(ham_bml, rho_bml, threshold, ncoeffs, kbt, ef, verbose)
+
+    implicit none
```

Comment:
Is that needed? I thought that when you specify the `implicit none` at the top it applies to the whole module.